pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginClassLoader.java

```diff
@@ -31,27 +32,53 @@
 public class PluginClassLoader extends URLClassLoader {
 
   private final ClassLoader _classLoader;
+  private Method _addUrlMethod = null;
 
   public PluginClassLoader(URL[] urls, ClassLoader parent) {
     super(urls, parent);
     _classLoader = PluginClassLoader.class.getClassLoader();
+    
+    /**
+     * ClassLoader in java9+ does not extend URLClassLoader.
+     * If the class is not found in the parent classloader,
+     * it will be found in this classloader via findClass().
+     *
+     * @see https://bit.ly/2WROm1s
+     */
+
+    if (_classLoader instanceof URLClassLoader) {
+      try {
+        _addUrlMethod = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
+      } catch (NoSuchMethodException e) {
+        //this should never happen
+        ExceptionUtils.rethrow(e);
+      }
+      
+      _addUrlMethod.setAccessible(true);
```

Comment:
Whilst this is common in the pinot codebase, doing things like this prevents use on JDK16+, creating obstacles to adoption of highly relevant JDK innovations (Vector API, foreign memory access, foreign function interface and so on).

Reply:
Any alternative suggestion @richardstartin? Seems like the issue surfaced in Pinot 0.8.0 and is breaking spark ingestion job.